partner_contact_in_several_companies/models/multi_contact.py

```diff
@@ -0,0 +1,179 @@
+                expression.AND(([('contact_type', '=', 'standalone')], args)),
+                [('other_contact_ids', 'in', attached_contacts.ids)],
+            ))
+        return super(ResPartner, self).search(args, offset=offset,
+                                              limit=limit, order=order,
+                                              count=count)
+
+    @api.model
+    def create(self, vals):
+        modified_self = self._basecontact_check_context('create')
+        if not vals.get('name') and vals.get('contact_id'):
+            vals['name'] = modified_self.browse(vals['contact_id']).name
+        return super(ResPartner, modified_self).create(vals)
+
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
+        modified_self = self._basecontact_check_context('read')
+        return super(ResPartner, modified_self).read(fields=fields, load=load)
+
+    @api.multi
+    def write(self, vals):
+        modified_self = self._basecontact_check_context('write')
+        return super(ResPartner, modified_self).write(vals)
+
+    @api.multi
+    def unlink(self):
+        modified_self = self._basecontact_check_context('unlink')
+        return super(ResPartner, modified_self).unlink()
+
+    @api.multi
+    def _commercial_partner_compute(self, name, args):
+        """ Returns the partner that is considered the commercial
+        entity of this partner. The commercial entity holds the master data
+        for all commercial fields (see :py:meth:`~_commercial_fields`) """
+        result = super(ResPartner, self)._commercial_partner_compute(name,
+                                                                     args)
+        for partner in self:
+            if partner.contact_type == 'attached' and not partner.parent_id:
+                result[partner.id] = partner.contact_id.id
+        return result
+
+    def _contact_fields(self):
+        """ Returns the list of contact fields that are synced from the parent
+        when a partner is attached to him. """
+        return ['name', 'title']
+
+    def _contact_sync_from_parent(self):
+        """ Handle sync of contact fields when a new parent contact entity
+        is set, as if they were related fields
+        """
+        self.ensure_one()
+        if self.contact_id:
+            contact_fields = self._contact_fields()
+            sync_vals = self._update_fields_values(self.contact_id,
+                                                   contact_fields)
+            self.write(sync_vals)
+
+    def update_contact(self, vals):
+        if self.env.context.get('__update_contact_lock'):
+            return
+        contact_fields = self._contact_fields()
+        contact_vals = dict(
+            (field, vals[field]) for field in contact_fields if field in vals
+        )
+        if contact_vals:
+            self.with_context(__update_contact_lock=True).write(contact_vals)
+
+    @api.model
+    def _fields_sync(self, partner, update_values):
```

Comment:
I don't know - it is not my module.  I just want to use it version 9, so I upgraded it.  I just need a working version accepted.